src/drivers/amd/vangogh/acp_dma.c

```diff
@@ -0,0 +1,261 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+//Copyright(c) 2023 AMD. All rights reserved.
+//
+//Author:       Basavaraj Hiregoudar <basavaraj.hiregoudar@amd.com>
+//              SaiSurya, Ch <saisurya.chakkaveeravenkatanaga@amd.com>
+
+#include <sof/audio/component.h>
+#include <platform/chip_registers.h>
+#include <platform/fw_scratch_mem.h>
+#include <platform/chip_offset_byte.h>
+#include <sof/drivers/acp_dai_dma.h>
+#include <rtos/timer.h>
+#include <rtos/alloc.h>
+#include <sof/lib/dma.h>
+#include <sof/lib/io.h>
+#include <sof/lib/notifier.h>
+#include <sof/lib/uuid.h>
+#include <sof/math/numbers.h>
+#include <sof/platform.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <platform/acp_dma.h>
+
+void dma_reconfig(struct dma_chan_data *channel, uint32_t bytes)
+{
+	uint32_t strt_idx = 0;
+	uint32_t src = 0;
+	uint32_t dest = 0;
+	uint32_t tail = 0, head = 0;
+	uint32_t src1 = 0;
+	uint32_t dest1 = 0;
```

Comment:
Initialization on the safer side.

Reply:
`head` seems local to the `for` loop so it  should be declared in the for loop and not leak to the entire function. This would avoid this initialisation debate entirely

Reply:
> Initialization on the safer side.

Apparently `dma_setup` does not need such "safety".